lib/java_buildpack/util/cache/cached_file.rb

```diff
@@ -33,7 +33,7 @@ class CachedFile
         # @param [String] uri a uri which uniquely identifies the file in the cache
         # @param [Boolean] mutable whether the cached file should be mutable
         def initialize(cache_root, uri, mutable)
-          key            = URI.escape(uri.sanitize_uri, ':/')
+          key            = URI.escape(uri.sanitize_uri, ':/&')
```

Comment:
I'm sorry I don't wanted to smuggle this into the core functionality with the PR. Hope this was okay tough. Next time I'll talk to you first - or create an issue.